felt/src/lib.rs

```diff
@@ -788,6 +788,16 @@ mod test {
             let p = &BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap();
             prop_assert!(&x.to_biguint() < p);
         }
+
+        #[test]
+        // Property-based test that ensures, for 100 felt values that are randomly generated each time tests are run, that a felt created using Felt::from_bytes_be doesn't fall outside the range [0, p].
+        // In this and some of the following tests, The value of {x} can be either [0] or a very large number, in order to try to overflow the value of {p} and thus ensure the modular arithmetic is working correctly.
+        fn from_bytes_be_in_range(ref x in "(0|[1-9][0-9]*)") {
+            let x = &Felt::from_bytes_be(x.as_bytes());
+            let p = &Felt::from(&(BigUint::parse_bytes(PRIME_STR[2..].as_bytes(), 16).unwrap() - 1_u32));
```

Comment:
I missed the `- 1` at the end. Now it makes sense. But note you already have the `Felt::max()` method for that. And calling it `p` is misleading, since it's __not__ the prime.